share/rpcauth/rpcauth.py

```diff
@@ -9,26 +9,36 @@
 import base64
 import hmac
 
-if len(sys.argv) < 2:
-    sys.stderr.write('Please include username as an argument.\n')
-    sys.exit(0)
+def generate_salt():
+    # This uses os.urandom() underneath
+    cryptogen = SystemRandom()
 
-username = sys.argv[1]
+    # Create 16 byte hex salt
+    salt_sequence = [cryptogen.randrange(256) for _ in range(16)]
+    return ''.join([format(r, 'x') for r in salt_sequence])
 
-#This uses os.urandom() underneath
-cryptogen = SystemRandom()
+"""Create 32 byte b64 password"""
```

Comment:
Docstrings should be the first statement *inside* the function (https://www.python.org/dev/peps/pep-0257/#what-is-a-docstring)

Reply:
Done